### PR TITLE
Slk 11764

### PR DESCRIPTION
**Pull request by dheater**
Propagate non-zero first row size not zero error for bitmaps from the library instead of panic'ing

Issue #34

### CODE REVIEW

File: ironrdp/src/basic_output/bitmap.rs [hunk 1]
```diff
@@ -97,7 +97,13 @@ impl<'a> PduBufferParsing<'a> for BitmapData<'a> {
...
 
         let compressed_data_header = if !compression_flags.contains(Compression::NOT_COMPRESSED) {
-            Some(CompressedDataHeader::from_buffer_consume(buffer)?)
+            let consumed_buffer = CompressedDataHeader::from_buffer_consume(buffer);
+            match consumed_buffer {
+                Ok(buffer) => Some(buffer),
+                Err(error) => {
+                    return Err(error);
+                },
+            }
```
Comment (CBenoit) [inline ironrdp/src/basic_output/bitmap.rs:106]: I think this is strictly equivalent to using the `?` operator as before. Am I missing something?

File: ironrdp/src/basic_output/fast_path.rs [hunk 2]
```diff
@@ -177,8 +177,14 @@ impl<'a> FastPathUpdate<'a> {
                 Ok(Self::SurfaceCommands(commands))
             }
             UpdateCode::Bitmap => {
-                let bitmap = Bitmap::from_buffer_consume(buffer)?;
-                Ok(Self::Bitmap(bitmap))
+                let bitmap = Bitmap::from_buffer_consume(buffer);
+                match bitmap {
+                    Ok(bitmap) => Ok(Self::Bitmap(bitmap)),
+                    Err(error) => {
+                        println!("Invalid bitmap: {:?}", error);
```
Comment (CBenoit) [inline ironrdp/src/basic_output/fast_path.rs:184]: We generally don't want libraries to `println!`

**Review by CBenoit — COMMENTED** [review]
Thank you!

Regarding issue https://github.com/Devolutions/IronRDP/issues/34

> Instead of panic, seems like we should handle this error in a way that allows the client to make a best effort to continue.

I don't see where the code was panicking

File: ironrdp/src/basic_output/fast_path.rs (hunk 2 above)
Comment (dheater) [reply ironrdp/src/basic_output/fast_path.rs:184]: Agree.  I forgot that one.

Comment (CBenoit) [reply ironrdp/src/basic_output/fast_path.rs:184]: Then I think it's good to go with
```rust
let bitmap = Bitmap::from_buffer_consume(buffer).map_err(|_| FastPathError::UnsupportedFastPathUpdate(code))?;
```
if the idea is just to override the Bitmap error :slightly_smiling_face:

File: ironrdp/src/basic_output/bitmap.rs (hunk 1 above)
Comment (dheater) [reply ironrdp/src/basic_output/bitmap.rs:106]: You're correct. This can just be allowed to propagate up.

Comment (dheater) [reply ironrdp/src/basic_output/bitmap.rs]: Reverted changes

File: ironrdp/src/basic_output/fast_path.rs (hunk 2 above)
Comment (dheater) [reply ironrdp/src/basic_output/fast_path.rs]: Yep. `map_err` is cleaner!

Change to propagating as `FastPathError::BitmapError` and handling it in the client.

**Review by CBenoit — APPROVED** [review]
This looks good to me! (@awakecoding if you have any comment, otherwise it's good to merge)

**Pull request merged**